pallets/asset/src/lib.rs

```diff
@@ -740,31 +751,36 @@ decl_module! {
             let mut issued_in_this_round = Self::issued_in_funding_round(&ticker_round);
 
             // A round of per-investor checks
-            for i in 0..issue_asset_items.len() {
+            for IssueAssetItem { investor_did, value } in &issue_asset_items {
                 ensure!(
-                    Self::check_granularity(&ticker, issue_asset_items[i].value),
+                    Self::check_granularity(&ticker, *value),
```

Comment:
Again not really specific to this PR, but my feeling is we'll need to refactor some of these batch functions to make it clearer as to what happens if it fails part way through. e.g. in this case, it looks like some of the batch may be processed, and some may not be. One option is to just move to using the utility batch functions everywhere. I'll create a separate story for this.

Reply:
(Also, the function is pretty large, so I'd suggest splitting it, if possible.)